### ISSUE TITLE
When passing empty string to AutoParse it panics

### ISSUE THREAD

**Issue by petergtz** [rickb777/date#4]
Hi,

I noticed that when passing `""` to `date.AutoParse`, it panics with the following error:

```
  runtime error: index out of range
  /usr/local/Cellar/go/1.12.5/libexec/src/runtime/panic.go:44

  Full Stack Trace
  	/usr/local/Cellar/go/1.12.5/libexec/src/runtime/panic.go:522 +0x1b5
  github.com/petergtz/alexa-journal/vendor/github.com/rickb777/date.AutoParse(0x0, 0x0, 0x0, 0x0, 0x0)
  	/Users/pego/workspace/go/src/github.com/petergtz/alexa-journal/vendor/github.com/rickb777/date/parse.go:39 +0x4e4
  github.com/petergtz/alexa-journal.DateFrom(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0)
```

Instead, I would expect it to return an error.

Thank you,
Peter

**Comment by rickb777** [comment]
Fixed in v1.12.4

**Comment by rickb777** [comment]
Thanks for raising this. :)
